From 340929b10cee5dd58a3e9d3907ebf332341e4612 Mon Sep 17 00:00:00 2001
From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>
Date: Thu, 2 Jun 2016 15:11:50 +0200
Subject: [PATCH 24/24] q35: Remove 7.0, 7.1, 7.2 machine types

RH-Author: Dr. David Alan Gilbert <dgilbert@redhat.com>
Message-id: <1464880310-29953-7-git-send-email-dgilbert@redhat.com>
Patchwork-id: 70544
O-Subject: [RHEL-7.3 qemu-kvm-rhev PATCH 6/6] q35: Remove 7.0, 7.1, 7.2 machine types
Bugzilla: 1342015
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Eduardo Habkost <ehabkost@redhat.com>

From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>

We're not promising migration compatibility between 7.[012]->7.3
and they were previously listed as tech preview.  To avoid confusion
remove them, forcing people to convert to the new maintained 7.3
or later.

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/i386/pc_q35.c | 97 --------------------------------------------------------
 1 file changed, 97 deletions(-)

diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index 1fe71b0..af29b3e 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -343,100 +343,3 @@ static void pc_q35_machine_rhel730_options(MachineClass *m)
 
 DEFINE_PC_MACHINE(q35_rhel730, "pc-q35-rhel7.3.0", pc_q35_init_rhel730,
                   pc_q35_machine_rhel730_options);
-
-static void pc_q35_compat_rhel720(MachineState *machine)
-{
-    pc_q35_compat_rhel730(machine);
-
-    savevm_skip_section_footers();
-    global_state_set_optional();
-}
-
-static void pc_q35_init_rhel720(MachineState *machine)
-{
-    pc_q35_compat_rhel720(machine);
-    pc_q35_init(machine);
-}
-
-static void pc_q35_machine_rhel720_options(MachineClass *m)
-{
-    pc_q35_machine_rhel730_options(m);
-    m->is_default = 0;
-    m->alias = NULL;
-    m->desc = "RHEL-7.2.0 PC (Q35 + ICH9, 2009)";
-    SET_MACHINE_COMPAT(m, PC_RHEL7_2_COMPAT);
-}
-
-DEFINE_PC_MACHINE(q35_rhel720, "pc-q35-rhel7.2.0", pc_q35_init_rhel720,
-                  pc_q35_machine_rhel720_options);
-
-static void pc_q35_compat_rhel710(MachineState *machine)
-{
-    PCMachineState *pcms = PC_MACHINE(machine);
-    PCMachineClass *pcmc = PC_MACHINE_GET_CLASS(pcms);
-
-    /* 7.1.0 is based on 2.1.2, 7.2.0 is based on 2.3 */
-    pc_q35_compat_rhel720(machine);
-
-    /* From pc_compat_2_2 */
-    pcmc->rsdp_in_ram = false;
-    machine->suppress_vmdesc = true;
-
-    /* From pc_compat_2_1 */
-    pcmc->enforce_aligned_dimm = false;
-    pcmc->smbios_uuid_encoded = false;
-    x86_cpu_change_kvm_default("svm", NULL);
-
-    /* From pc_q35_2_4_machine_options */
-    pcmc->broken_reserved_end = true;
-}
-
-
-static void pc_q35_init_rhel710(MachineState *machine)
-{
-    pc_q35_compat_rhel710(machine);
-    pc_q35_init(machine);
-}
-
-static void pc_q35_machine_rhel710_options(MachineClass *m)
-{
-    m->family = "pc_q35_Z";
-    m->desc = "RHEL-7.1.0 PC (Q35 + ICH9, 2009)";
-    m->default_machine_opts = "firmware=bios-256k.bin";
-    SET_MACHINE_COMPAT(m, PC_RHEL7_1_COMPAT);
-}
-
-DEFINE_PC_MACHINE(q35_rhel710, "pc-q35-rhel7.1.0", pc_q35_init_rhel710,
-                  pc_q35_machine_rhel710_options);
-
-static void pc_q35_compat_rhel700(MachineState *machine)
-{
-    PCMachineState *pcms = PC_MACHINE(machine);
-    PCMachineClass *pcmc = PC_MACHINE_GET_CLASS(pcms);
-    pc_q35_compat_rhel710(machine);
-
-    /* Upstream enables it for everyone, we're a little more selective */
-    x86_cpu_change_kvm_default("x2apic", NULL);
-
-    pcmc->smbios_legacy_mode = true;
-    pcmc->has_reserved_memory = false;
-    migrate_cve_2014_5263_xhci_fields = true;
-    global_state_set_optional();
-}
-
-static void pc_q35_init_rhel700(MachineState *machine)
-{
-    pc_q35_compat_rhel700(machine);
-    pc_q35_init(machine);
-}
-
-static void pc_q35_machine_rhel700_options(MachineClass *m)
-{
-    m->family = "pc_q35_Z";
-    m->desc = "RHEL-7.0.0 PC (Q35 + ICH9, 2009)";
-    m->default_machine_opts = "firmware=bios-256k.bin";
-    SET_MACHINE_COMPAT(m, PC_RHEL7_0_COMPAT);
-}
-
-DEFINE_PC_MACHINE(q35_rhel700, "pc-q35-rhel7.0.0", pc_q35_init_rhel700,
-                  pc_q35_machine_rhel700_options);
-- 
1.8.3.1

